### PR TITLE
feat(ui5-input) implement valueStateMessage when no suggestions

### PR DESCRIPTION
**Pull request by ivoplashkov**
Implement valueStateMessage when the input has no suggestions.

### CODE REVIEW

File: packages/main/src/Input.js [hunk 1]
```diff
@@ -407,6 +433,13 @@ class Input extends UI5Element {
 			}
 		}
 
+		if (!this.firstRendering && !this.Suggestions && this._getPopover() && this.hasValueStateMessage) {
+			this.toggle(this.shouldDisplayOnlyValueStateMessage);
+			if (this._getPopover().contentDOM) {
+				this._getPopover().contentDOM.style.display = "none";
```
Comment (vladitasev) [inline packages/main/src/Input.js]: Do not directly access the DOM unless absolutely necessary. Change some state of the component that reflects in the "get styles()" and uses the template to update the styles. In order words, don't change DOM directly, only through the template. The reason is that you introduce hidden state that will be hard to track in the future

File: packages/main/src/Input.js [hunk 2]
```diff
@@ -508,6 +545,12 @@ class Input extends UI5Element {
 		}
 	}
 
+	_handleResize() {
+		if (this.hasValueStateMessage) {
+			this._getPopover().header[0].style.width = `${this.offsetWidth}px`;
```
Comment (vladitasev) [inline packages/main/src/Input.js]: Same, this should be a metadata private property, which will invalidate the input and set the style in the template

File: packages/main/src/Input.js [hunk 3]
```diff
@@ -526,6 +569,28 @@ class Input extends UI5Element {
 		}
 	}
 
+	toggle(bToggle) {
```
Comment (vladitasev) [inline packages/main/src/Input.js]: we don't use hungarian notation, just overwrite the parameter on the next line, no need for another variable

File: packages/main/src/themes/ValueStateMessage.css [hunk 4]
```diff
@@ -0,0 +1,26 @@
+.ui5-input-valuestatemessage-root {
```
Comment (vladitasev) [inline packages/main/src/themes/ValueStateMessage.css]: Use parameters for color, font family, etc...

File: packages/main/src/Input.js (hunk 1 above)
Comment (ivoplashkov) [reply packages/main/src/Input.js]: Done.

File: packages/main/src/Input.js (hunk 2 above)
Comment (ivoplashkov) [reply packages/main/src/Input.js]: Done.

File: packages/main/src/Input.js (hunk 3 above)
Comment (ivoplashkov) [reply packages/main/src/Input.js]: Done.

File: packages/main/src/themes/ValueStateMessage.css (hunk 4 above)
Comment (ivoplashkov) [reply packages/main/src/themes/ValueStateMessage.css]: Done.

File: packages/main/src/Input.js [hunk 5]
```diff
@@ -526,6 +576,40 @@ class Input extends UI5Element {
 		}
 	}
 
+	toggle(isToggled) {
+		if (isToggled) {
+			this.openPopover();
+		}
```
Comment (ilhan007) [inline packages/main/src/Input.js]: You can be sure that in the "else" statement "isToggled" would be false, so you can skip checking it.
And, use "isOpen", instead if "this._isPopoverOpen".

And, better 
```js
if (isToggled) {
		this.openPopover();
} else {
	    this.closePopover(); // if you want you can make the check for isOpen in the "closePopover"
}
```

File: packages/main/src/Input.js [hunk 6]
```diff
@@ -691,10 +776,52 @@ class Input extends UI5Element {
...
+	get shouldDisplayDefaultValueStateMessage() {
+		return !this.valueStateMessage.length && this.hasValueStateMessage;
+	}
+
 	get hasValueState() {
 		return this.valueState !== ValueState.None;
 	}
 
+	get hasValueStateMessage() {
+		return this.valueStateMessage && this.hasValueState && this.valueState !== ValueState.Success;
```
Comment (ilhan007) [inline packages/main/src/Input.js]: this.valueStateMessage is always truthy, I think you can remove that check or if you need it you should check for the length.

**Pull request merged**